source-oracle/discovery.go

```diff
@@ -109,37 +109,36 @@ func translateRecordField(column *sqlcapture.ColumnInfo, val interface{}) (inter
 	case nil:
 		return nil, nil
 	case string:
-		if dataType.jsonType != "string" {
-			var out = reflect.New(dataType.t).Interface()
-			if err := json.Unmarshal([]byte(v), out); err != nil {
-				return nil, err
-			}
-			return out, nil
+		if dataType.JsonType == "integer" {
+			return strconv.Atoi(v)
+		} else if dataType.JsonType == "number" {
+			return strconv.ParseFloat(v, 64)
```

Comment:
Do a more direct conversion from string to integer or float. The previous approach was more error-prone